xml/hwbp_registers.xml

```diff
@@ -276,14 +276,21 @@
             and writing it has no effect.
         </field>
         <field name="select" bits="19" access="WARL" reset="0">
+            This bit determines the contents of the XLEN-bit value to be
+            compared.
+
             0: Perform a match on the lowest virtual address of the access.  In
             addition, it is recommended that the trigger also matches if any of
             the other accessed virtual addresses match.
             (E.g. on a 32-bit read from 0x4000, the lowest address is 0x4000
             and the other addresses are 0x4001, 0x4002, and 0x4003.)
+            If the virtual address is smaller than DXLEN, the high bits
+            contain 0.
```

Comment:
This sentence isn't clear to me.  I think that you mean that if XLEN<DXLEN then address[DXLEN-1:XLEN]=0 for purposes of the comparison.  But I don't think this works for the case where DXLEN=64 and MXLEN=32 and M mode sets up a trigger with action!=1.  The M-mode write to tdata2 is defined to sign-extend [63:32].  If address[63:32]=0 always but tdata[63:32] is sign-extended then you can't match on any address with [31]=1.

The other two options are:
1. Sign-extend the address before comparison.  But if you have MXLEN=64 and SXLEN=32 and satp.MODE=Bare then the top byte of what S mode can access is VA=PA=0x00000000FFFFFFFF, not VA=PA=0xFFFFFFFFFFFFFFFF (which is likely a vacant PMA region).

2. Say that no bits above the current XLEN are compared.  In the MXLEN=64 and SXLEN=32 case, this means that you will ignore tdata2[63:32] which can also be weird because address 0x12345678 matches against tdata2=0xfeedface12345678.

I bet that 2 would be more popular with implementations so they don't have to have 33 bits fanout from addr[31] (for no real benefit).

Reply:
Option 2 makes sense to me. The quirk you point out is perhaps a little odd, but at the same time it doesn't make sense to include bits in the comparison which cannot be part of the virtual address.